packages/pigeon/lib/java_generator.dart

```diff
@@ -113,22 +100,6 @@ void _writeCodec(Indent indent, Api api) {
   });
 }
 
-/// This performs Dart to Java type conversions.  If performs a passthrough of
-/// the input if it can't be converted.
-// TODO(gaaclarke): Remove this method and unify it with `_javaTypeForDartType`.
```

Comment:
Why not do this unification? It looks the large majority of calls to _javaTypeForDartType want this behavior, and are now individually re-implementing it (which is both verbose, and error-prone because it repeats the argument twice).

I see that you need _javaTypeForDartType to default to the existing behavior since it's used an function argument to another method, but you should be able to add a `{bool allowCustomTypes = false}` and then just change the one line of the final return to be conditional on that. It would effectively give you `_javaTypeForDartTypePassthrough` but with no duplication of code.

(If the issue is the nullability of the return, you could explicitly document that it can't return null if that's true, and then make a `_javaTypeForDartOrCustomType(TypedEntity entity) => _javaTypeForDartType(entity, allowCustomTypes: true)!`

Reply:
Yep, had the same thought, done.  The comment was referring to some duplication of logic that we did get rid of.